docs/argo-core-concepts.md

```diff
@@ -0,0 +1,146 @@
+
+# Argo Basic Concepts
```

Comment:
can we just call this `core-concepts.md`?

Reply:
Sure